examples/java-jsp/README.md

```diff
@@ -0,0 +1,152 @@
+
+To install Maven, run the following command:
+
+```bash
+# apt-get install maven
+```
+
+### Tomcat
+
+To install tomcat locally, you could run the following sequence of commands,
+sourced from
+[Digital Ocean's Tutorial](https://www.digitalocean.com/community/tutorials/install-tomcat-9-ubuntu-1804)
+
+Create `tomcat` user and group.
+```bash
+# groupadd tomcat
+# useradd -s /bin/false -g tomcat -d /opt/tomcat tomcat
+```
+
+Download and extract `tomcat` to `/opt/tomcat`.
+```bash
+$ curl -O https://downloads.apache.org/tomcat/tomcat-9/v9.0.43/bin/apache-tomcat-9.0.43.tar.gz
+# mkdir /opt/tomcat
+# tar xzvf apache-tomcat-*tar.gz -C /opt/tomcat --strip-components=1
+```
+
+Set permissions and groups.
+```bash
+# chgrp -R tomcat /opt/tomcat
+# chmod -R g+r /opt/conf
+# chmod g+x /opt/conf
+# chown -R tomcat /opt/webapps /opt/work /opt/temp /opt/logs
+```
+
+Find out where your JAVA_HOME is located.
+```bash
+```
+
+Create `tomcat` service file at `/etc/systemd/system/tomcat.service`.
+```bash
+[Unit]
+Description=Apache Tomcat Web Application Container
+After=network.target
+
+[Service]
+Type=forking
+
+Environment=JAVA_HOME=/usr/lib/jvm/java-1.11.0-openjdk-amd64
+Environment=CATALINA_PID=/opt/tomcat/temp/tomcat.pid
+Environment=CATALINA_HOME=/opt/tomcat
+Environment=CATALINA_BASE=/opt/tomcat
+Environment='CATALINA_OPTS=-Xms512M -Xmx1024M -server -XX:+UseParallelGC'
+Environment='JAVA_OPTS=-Djava.awt.headless=true -Djava.security.egd=file:/dev/./urandom'
+
+ExecStart=/opt/tomcat/bin/startup.sh
+ExecStop=/opt/tomcat/bin/shutdown.sh
+
+User=tomcat
+Group=tomcat
+UMask=0007
+RestartSec=10
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Reload the systemd daemon and start the service:
+```bash
+# systemctl daemon-reload
+# systemctl start tomcat
+# systemctl status tomcat
+```
+
+Optionally, enable it at start-up with: 
+```bash
+# systemctl enable tomcat
+```
+
+### Build DIDKit
+
+The web application makes use of DIDKit to handle credentials and presentations,
+please refer to the project's documentation to build the library for your platform,
+`libdidkit.so` on UNIX-like systems, `didkit.dll` on Windows, `libdidkit.dylib`
+on MacOS, etc.
+
+Then you will have to add it to the classpath of your platform. On UNIX systems,
+for example, you can copy (or symlink) `libdidkit.so` to `/usr/lib` or
+`/usr/local/lib`. In the instructions below, we will list commands to create a
+symlink to the local build folder.
+
+You will then require the Java artifact (`didkit.jar`). This example project
+already has a symlink in the build directory.  All you have to do is run the
+following commands in the root folder to build everything and ensure proper
+linking:
+
+```bash
+$ cargo build
+# Use libdidkit.so for UNIX-like, didkit.dll for Windows, libdidkit.dylib for MacOS
+$ make -C lib ../target/didkit.jar
+```
+
+## Install `didkit.jar` to local Maven repository
+
+To avoid a complicated Maven setup, the easiest way to be able to include the
```

Comment:
At this point, we should instruct the user to change directories to `examples/java-jsp`.